app/addons/activetasks/components.react.jsx

```diff
@@ -0,0 +1,387 @@
+        progressMessage.push('Progress: ' + model.get('progress') + '%');
+      }
+
+      if (type === 'indexer') {
+        progressMessage.push('Processed ' + model.get('changes_done') + ' of ' + model.get('total_changes') + ' changes.');
+      } else if (type === 'replication') {
+        progressMessage.push(model.get('docs_written')+ ' docs written.');
+        if (!_.isUndefined(model.get('changes_pending'))) {
+          progressMessage.push(model.get('changes_pending') + ' pending changes.');
+        }
+      }
+      if (!_.isUndefined(model.get('source_seq'))) {
+        progressMessage.push('Current source sequence: ' + model.get('source_seq') + '. ');
+      }
+      if (!_.isUndefined(model.get('changes_done'))) {
+        progressMessage.push(model.get('changes_done') + ' Changes done.');
+      }
+      
+      return progressMessage;
+    },
+    passesFilter: function (item) {
+      var searchTerm = this.props.searchTerm;
+      var regex = new RegExp(searchTerm, 'g');
+      
+      var itemDatabasesTerm = '';
+      if (item.has('database')) {
+        itemDatabasesTerm += item.get('database'); 
+      }
+      if (item.has('source')) {
+        itemDatabasesTerm += item.get('source'); 
+      }
+      if (item.has('target')) {
+        itemDatabasesTerm += item.get('target'); 
+      }
+
+      if (regex.test(itemDatabasesTerm)) {
+        return this.passesTabFilter(item.get('type'));
+      }
+      return false;
+    },
+    passesTabFilter: function (type) {
+      if (this.props.tab.toLowerCase() === type.replace('_', ' ') || this.props.tab ==='All Tasks') {
+        return true;
+      }
+      return false;
+    },
+    sortByHeader: function (e) {
+      var headerField = $(e.currentTarget),
+      columnName = headerField.attr('data-type');
+      Actions.sortByColumnHeader(columnName);
+    },
+    render: function () {
+      var collection = this.props.collection;
+      var resultsCount = 0;
+
+      return (
+        <div id="dashboard-lower-content">
+          <table className="table table-bordered table-striped active-tasks">
+            <thead>
+              <tr>
+                <th className="type"      data-type="type"        onClick={this.sortByHeader}> Type</th>
+                <th className="database"  data-type="database"    onClick={this.sortByHeader}> Database</th>
+                <th className="started"   data-type="started_on"  onClick={this.sortByHeader}> Started on</th>
+                <th className="updated"   data-type="updated_on"  onClick={this.sortByHeader}> Last updated on</th>
+                <th className="pid"       data-type="pid"         onClick={this.sortByHeader}> PID</th>
+                <th className="status"    data-type="progress"    onClick={this.sortByHeader}> Status</th>
+              </tr>
+            </thead>
+            <tbody className="js-tasks-go-here">
+              { collection.map(function (item, iteration) {
```

Comment:
This should be done in a a separate function. Here is an [example](https://github.com/apache/couchdb-fauxton/blob/master/app/addons/cors/components.react.jsx#L132)